### PR TITLE
Use C++ system headers

### PR DESCRIPTION
**Pull request by kklobe**
# Description

A long-overdue chore, I went and replaced the old C library .h files with the C++ equivalents.

I left `src/libs` alone for now, and the reelmagic `mgeg_decoder.h` as well (open to discussion).

As an interesting note, the C++ headers provide additional functionality in some cases. For example, Apple's Xcode 15.1 `ctime` provides the following:

```cpp
// TODO: Remove once rdar://101515782 is resolved
#ifdef _LIBCPP_ON_SEP
inline size_t strftime(char *, size_t, const char *, const struct tm *) { return 0; }
#endif
```

`cstring` provides:

```cpp
inline _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX14 size_t __constexpr_strlen(const char* __str) {
  // GCC currently doesn't support __builtin_strlen for heap-allocated memory during constant evaluation.
  // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=70816
#ifdef _LIBCPP_COMPILER_GCC
  if (__libcpp_is_constant_evaluated()) {
    size_t __i = 0;
    for (; __str[__i] != '\0'; ++__i)
      ;
    return __i;
  }
#endif
  return __builtin_strlen(__str);
}
```

# Manual testing

Ran the build artifacts for:

- macOS x86-64
- macOS arm64
- Windows MSYS2 x86-64

# Checklist

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [ ] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [ ] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

**Comment by kcgen** [comment]
Now that's some cleanup, and a nice runway to C++20 :airplane: 
(My guess is we'll queue this for 0.82, but I don't see anything risky either, so very well could be a trivial backport too. @johnnovak , your call)

**Review by kcgen — APPROVED** [review]

**Comment by kklobe** [comment]
> Now that's some cleanup, and a nice runway to C++20 ✈️ (My guess is we'll queue this for 0.82, but I don't see anything risky either, so very well could be a trivial backport too. @johnnovak , your call)

That was my main motivation, can't see any really good reason to risk some weird compiler-library-specific regressions for 0.81, but up to you guys.

**Comment by johnnovak** [comment]
> > Now that's some cleanup, and a nice runway to C++20 ✈️ (My guess is we'll queue this for 0.82, but I don't see anything risky either, so very well could be a trivial backport too. @johnnovak , your call)
> 
> That was my main motivation, can't see any really good reason to risk some weird compiler-library-specific regressions for 0.81, but up to you guys.

I see it as too noisy & too risky. It's a good long-term cleanup, no doubt, but very little to be gained from backporting it short-term. I'd say, 0.82 because I like my life simple 😎

**Review by johnnovak — APPROVED** [review]
Nice one @kklobe. I've done a similar cleanup in one of the files in my VGA CRTC commit, but I'm leaning towards holding this off until 0.82 as this is doing the same thing at a mass-scale.

**Pull request merged**
